docs/constraints.md

```diff
@@ -0,0 +1,88 @@
+# Introduction
+
+The purpose of this document is to introduce well-defined terminology to talk about different syntactic properties that an RDF graph with triple terms may possess. These properties are independent of one another and capture different types of restrictions that the WG may combine into a notion of well-formedness, either by including all of these properties or only a selected subset.
+
+# Preliminaries
+
+The definitions in this document consider a version of the abstract data model of RDF that is extended with triple terms in the following way.
+
+**Definition:** An **RDF graph** is a set of RDF triples.
+
+**Definition:** An **RDF triple** (usually called "triple") is a 3-tuple (*s*, *p*, *o*) where:
+
+* *s* is an IRI, a blank node, or a triple term,
+* *p* is an IRI, and
+* *o* is an IRI, a blank node, a literal, or a triple term.
+
+**Definition:** An **triple term** is a 3-tuple that is defined recursively as follows:
+
+* If *s* is an IRI or a blank node, *p* is an IRI, and *o* is an IRI, a blank node, or a literal, then (*s*, *p*, *o*) is a triple term.
+* If *s* is an IRI or a blank node, *p* is an IRI, and *o* is a triple term, then (*s*, *p*, *o*) is a triple term.
+* If *s* is a triple term, *p* is an IRI, and *o* is an IRI, a blank node, or a literal, then (*s*, *p*, *o*) is a triple term.
```

Comment:
Is there a reason this not also true?
```
* If *s* is a triple term, *p* is an IRI, and *o* is an IRI, a blank node, a literal, or a triple term, then (*s*, *p*, *o*) is a triple term.
```

Reply:
@TallTed Indeed, I forgot to cover this case in the definition. Thanks for pointing it out. I have fixed this issue in PR #118